File: axon-spring-boot/src/main/java/com/example/entity/Account.java
package com.example.entity;

import com.example.aggregates.Status;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "accounts")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Account {
    @Id
    private String id;
    private Double accountBalance;
    private String currency;
    @Enumerated(value = EnumType.STRING)
    private Status status;
}

File: spring-boot-kotlin-starter/src/main/kotlin/com/example/entity/Account.kt
package com.example.entity

import javax.persistence.*

@Entity
@Table(name = "accounts")
data class Account(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null,
        var name: String? = null
) {
    @OneToMany(cascade = [(CascadeType.ALL)], fetch = FetchType.LAZY, orphanRemoval = true)
    @JoinColumn(name = "account_id")
    var articles: MutableList<Article> = mutableListOf()

    fun addArticle(article: Article) {
        this.articles.add(article)
    }
}
